src/backend/distributed/executor/executor.c

```diff
@@ -348,6 +361,17 @@ CitusExecScan(CustomScanState *node)
 		if (MultiShardConnectionType == SEQUENTIAL_CONNECTION)
 		{
 			targetPoolSize = 1;
+
+			/* we favor MultiShardConnectionType over ForceMaxQueryParallelization */
+			ForceMaxQueryParallelization = false;
```

Comment:
either (a) create a local copy of `ForceMaxQueryParallelization ` in the execution struct. Only set this locally in the execution (b) remove this setting here and act accordingly when  `MultiShardConnectionType == SEQUENTIAL_CONNECTION`

Otherwise, it is easy to mass things up by changing both settings at the same time.